lib/src/elastic_client_impl.dart

```diff
@@ -102,15 +102,16 @@ class Client {
     return rs.bodyAsMap['deleted'] as int ?? 0;
   }
 
-  Future<SearchResult> search(String index, String type, Map query,
-      {int offset, int limit, bool fetchSource = false, Map suggest}) async {
+  Future<SearchResult> search(String index, String type, Map<String, dynamic> query,
+      {int offset, int limit, bool fetchSource = false, Map suggest, Map aggs, bool rawQuery = false}) async {
     final path = [index, type, '_search'];
-    final map = {
+    final map = rawQuery ? query: {
```

Comment:
why do you need the `rawQuery`?

Reply:
In the latest versions of Elastic Database the query format change, and with your hardCoded {"config": query} will be common to be unable to use in that way.

I have passed by some problems with the query serialization and the deserialization of the return with differentes elastic server versions.
This situations has obrigate me to change your elastic_client code to be possible to use.

Another suggestion is to be possible to use a rawQuery and be possible to has the rawReturn.

Reply:
Hm, in that case, wouldn't it be better to make it as a client-level flag that you set at the constructor? I fear that passing it in each `query()` makes it cumbersome...

With that, how does logic change if you do have `rawQuery = true` and `offset` + `limit` specified? Could you also handle that case?

If you can copy here the query objects that were working and no longer working, I'd help with the code too.

Reply:
Look my last commit, the total_count return in elastic v7.0 its a Map.
Think that in the future the query will have more than aggregations, limit, and other things, and the user want to build your own query.

Should be cool if the user can make your own map of query and send it as raw and dynamic map and get the response as a dynamic json. To treat like he want.

Reply:
I see it now, okay. Couple of things before I can merge it, first of all:

What do you think: would it make sense to have a separate `searchRaw` (or `rawSearch`) method? The current search would invoke it with the current query construction, while other could call it directly. That way the `bool rawQuery` would not be necessary.

The alternative is to also check if `rawQuery=true` and any other param is specified. E.g. `rawQuery=true` and `limit = 10` should throw an error to the client (and not swallow the lost parameter silently).

Reply:
+1 for having this as a separate rawSearch method

Reply:
+1 for having this as a separate rawSearch method

Reply:
nice. I just modified everything and added a load more queries and removed a lot of the doc stuff for the new versions.

https://github.com/elgansayer/elastic_client

Tbh i should probably clean it up and submit a PR.

Reply:
@elgansayer: I'd encourage you to make that PR!